### PR TITLE
Golden Doc Updates

### PR DESCRIPTION
**Pull request by Piinks**
## Description

This adds further documentation that relates to #38473, including sample test output.
~Waiting on assets to land in https://github.com/flutter/assets-for-api-docs/pull/101 to embed with documentation.~

## Related Issues

Follow-up doc improvements to #38473

## Tests

Docs only

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by Piinks** [comment]
> It looks like this pull request may not have tests. Please make sure to add tests before merging. While there are exceptions to this rule, if this patch modifies code it is probably not an exception.
> 
> **Reviewers**: Read the [Tree Hygiene page](https://github.com/flutter/flutter/wiki/Tree-hygiene#how-to-review-code) and make sure this patch meets those guidelines before LGTMing.

Doc change only. :)

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter_test/lib/src/goldens.dart [hunk 1]
```diff
@@ -23,6 +23,33 @@ import 'package:test_api/test_api.dart' as test_package show TestFailure;
...
+/// prevent unintended changes as you develop by testing against them.
+///
+/// Here, a minor code change has altered the appearance of a widget. A golden
+/// file test has compared the image generated at the time of the test to the
+/// golden master file that was generated earlier. The test has identified the
+/// change, preventing unintended modifications.
+///
+/// |  Sample     |  Image |
+/// |-------------|--------|
+/// |  Master     | ![A golden master image](https://flutter.github.io/assets-for-api-docs/assets/flutter-test/goldens/widget_masterImage.png)  |
```
Comment (goderbauer) [inline packages/flutter_test/lib/src/goldens.dart]: Maybe say "Golden Master Image" here?

File: packages/flutter_test/lib/src/goldens.dart [hunk 2]
```diff
@@ -23,6 +23,33 @@ import 'package:test_api/test_api.dart' as test_package show TestFailure;
...
+/// |  Difference | ![The pixel difference](https://flutter.github.io/assets-for-api-docs/assets/flutter-test/goldens/widget_isolatedDiff.png)  |
+/// |  Test       | ![Test image](https://flutter.github.io/assets-for-api-docs/assets/flutter-test/goldens/widget_testImage.png) |
```
Comment (goderbauer) [inline packages/flutter_test/lib/src/goldens.dart]: maybe: "Test image after code modification"?

File: packages/flutter_test/lib/src/goldens.dart (hunk 1 above)
Comment (Piinks) [inline packages/flutter_test/lib/src/goldens.dart]:
```suggestion
/// |  Golden Master Image     | ![A golden master image](https://flutter.github.io/assets-for-api-docs/assets/flutter-test/goldens/widget_masterImage.png)  |
```

File: packages/flutter_test/lib/src/goldens.dart (hunk 2 above)
Comment (Piinks) [inline packages/flutter_test/lib/src/goldens.dart]:
```suggestion
/// |  Test image after modification       | ![Test image](https://flutter.github.io/assets-for-api-docs/assets/flutter-test/goldens/widget_testImage.png) |
```

**Pull request merged**